src/components/UserDetailsProps.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate, useParams, Link } from 'react-router-dom';
+import { Button, Card, Col, Image, Row, ListGroup } from 'react-bootstrap';
+import apiClient from '../services/api-client';
+import { FetchUserProps } from './Form';
+import '../index.css';
+import 'bootstrap/dist/css/bootstrap.css';
+import { FaCheck } from 'react-icons/fa';
+import { BsFillXCircleFill } from 'react-icons/bs';
+
+
+interface UserDetailsData extends FetchUserProps {
+  name: string;
+  company: string;
+  blog: string;
+  location: string;
+  email: string;
+  hireable: boolean;
+  bio: string;
+  twitter_username: string;
+  public_repos: number;
+  public_gists: number;
+  followers: number;
+  following: number;
+  created_at: number;
+  updated_at: number;
+  repos: string[];
+}
+
+const UserDetails = () => {
+  const { username } = useParams();
+  const [userDetails, setUserDetails] = useState<UserDetailsData>();
+  const [userRepos, setUserRepos] = useState<UserDetailsData[]>([]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const { data } = await apiClient.get(`/users/${username}`);
+        setUserDetails(data);
+      } catch {
+        throw new Error('Error fetching data');
+      }
+    };
+    const fetchUserRepos = async () => {
+      try {
+        const { data } = await apiClient.get(`/users/${username}/repos?per_page=5`);
+        setUserRepos(data);
+      } catch {
+        throw new Error('Error fetching data');
+      }
+    };
```

Comment:
These two functions do the same thing. Group them together like this and delete one
```suggestion
    const fetchData = async () => {
      try {
        const { data: userData } = await apiClient.get(`/users/${username}`);
        const { data: userReposData } = await apiClient.get(`/users/${username}/repos?per_page=5`);
        setUserDetails(userData);
        setUserRepos(userReposData);
      } catch {
        throw new Error('Error fetching data');
      }
    };
```